internal/storage/image.go

```diff
@@ -781,57 +770,28 @@ func (svc *imageService) ResolveNames(systemContext *types.SystemContext, imageN
 		return nil, err
 	}
 
-	domain, remainder := splitDockerDomain(imageName)
-	if domain != "" {
-		// this means the image is already fully qualified
-		registry, err := sysregistriesv2.FindRegistry(systemContext, imageName)
-		if err != nil {
-			return nil, err
-		}
-		if registry != nil && registry.Blocked {
-			return nil, fmt.Errorf("cannot use %q because it's blocked", imageName)
-		}
-		return imageNamesWithDigestOrTag([]string{imageName})
-	}
-	UnqualifiedSearchRegistries, err := sysregistriesv2.UnqualifiedSearchRegistries(systemContext)
+	resolved, err := shortnames.Resolve(systemContext, imageName)
 	if err != nil {
 		return nil, err
 	}
-	// we got an unqualified image here, we can't go ahead w/o registries configured
-	// properly.
-	if len(svc.lookup.UnqualifiedSearchRegistries) == 0 && len(UnqualifiedSearchRegistries) == 0 {
-		return nil, ErrNoRegistriesConfigured
-	}
-	// this means we got an image in the form of "busybox"
-	// we need to use additional registries...
-	// normalize the unqualified image to be domain/repo/image...
-	images := []string{}
-	for _, r := range append(svc.lookup.UnqualifiedSearchRegistries, UnqualifiedSearchRegistries...) {
-		rem := remainder
-		if r == "docker.io" && !strings.ContainsRune(remainder, '/') {
-			rem = "library/" + rem
-		}
-		image := r + "/" + rem
-		registry, err := sysregistriesv2.FindRegistry(systemContext, image)
-		if err != nil {
-			return nil, err
-		}
-		if registry != nil && registry.Blocked {
-			continue
-		}
-		images = append(images, image)
+
+	if desc := resolved.Description(); len(desc) > 0 {
+		logrus.Info(desc)
 	}
-	if len(images) == 0 {
-		return nil, fmt.Errorf("all search registries for %q are blocked", remainder)
+
+	images := make([]string, len(resolved.PullCandidates))
+	for i := range resolved.PullCandidates {
+		images[i] = resolved.PullCandidates[i].Value.String()
 	}
+
 	return imageNamesWithDigestOrTag(images)
```

Comment:
This is the only remaining caller; so please update it to work with `resolved.PullCandidates[].Value` directly instead of the current round-trip via `….Value.String()`.